src/espIdf/unitTest/testExecution.ts

```diff
@@ -0,0 +1,275 @@
+/*
+ * Project: ESP-IDF VSCode Extension
+ * File Created: Wednesday, 19th July 2023 12:05:38 pm
+ * Copyright 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { copy, pathExists, readFile, writeFile } from "fs-extra";
+import { join } from "path";
+import {
+  CancellationToken,
+  ShellExecution,
+  ShellExecutionOptions,
+  TaskPanelKind,
+  TaskPresentationOptions,
+  TaskRevealKind,
+  TaskScope,
+  Uri,
+  extensions,
+} from "vscode";
+import { ESP } from "../../config";
+import { readParameter } from "../../idfConfiguration";
+import { appendIdfAndToolsToPath, startPythonReqsProcess } from "../../utils";
+import { buildCommand } from "../../build/buildCmd";
+import { verifyCanFlash } from "../../flash/flashCmd";
+import { flashCommand } from "../../flash/uartFlash";
+import { jtagFlashCommand } from "../../flash/jtagCmd";
+import { OutputChannel } from "../../logger/outputChannel";
+import { parseStringPromise } from "xml2js";
+import { TaskManager } from "../../taskManager";
+
+export async function copyTestAppProject(
+  workspaceFolder: Uri,
+  testComponents: string[]
+) {
+  let unityAppDir: string = join(
+    extensions.getExtension(ESP.extensionID).extensionPath,
+    "templates",
+    "unity-app"
+  );
+  let destUnityAppDir = Uri.joinPath(workspaceFolder, "unity-app");
+  await copy(unityAppDir, destUnityAppDir.fsPath);
+  await updateTestComponents(destUnityAppDir, testComponents);
+  return destUnityAppDir;
+}
+
+export async function updateTestComponents(
```

Comment:
Also here to add a try catch since we are performing file read and write operations.